doc/headset.md

```diff
@@ -0,0 +1,430 @@
+            break;
+        default:
+        // console.log('some other event');
+}
+
+```
+
+### Methods
+
+#### `updateAudioInputDevice()`
+Function that will update what the active mic is in the headset library. This passes the newly selected mic label into the headset library where it will check to see which (if any) of the currently supported implementations will be a best fit for the new device This function should be called any time a new audio device is selected.
+
+This function is currently called from `client.ts` if a new audio device is selected through the function `sdk.updateDefaultDevices()`.
+
+This function will call through `sdk.media.findCachedDeviceByIdAndKind(newMicId)` which takes in the newly selected device's ID and will return the device's complete information including the necessary device label
+
+After receiving the complete device information, it will send the device's label to the headset library through `this.headsetLibrary.activeMicChange(completeDeviceInfo.label.toLowerCase())`
+
+Declaration:
+``` ts
+    updateAudioInputDevice (newMicId: string) : void;
+```
+
+Params:
+* `newMicId: string` - The ID of the newly selected device sent from the `client.ts`
+
+Returns: void
+
+#### `getCurrentSelectedImplementation()`
+Function that fetches the currently selected implementation from the headset library. This will help with checking proper flags for various functionality such as `this.getCurrentSelectedImplementation().isConnected`
+
+Declaration:
+``` ts
+    getCurrentSelectedImplementation () : VendorImplementation;
+```
+
+Params: none
+
+Returns: an instance of a VendorImplementation from the headset library
+
+#### `showRetry()`
+Function that determines if a "show retry" button should be shown on screen to allow the user to attempt to reconnect to their selected vendor and the corresponding implementation. The show retry button should show up if:
+    1.) The selected implementation has disabled retry capabilities
+    2.) The selected device is not supported (`!selectedImplementation`)
+    3.) The selected implementation is not connected (`!selectedImplementation.isConnected`)
+    4.) The selected implementation is not connecting (`!selectedImplementation.isConnecting`)
+
+Declaration:
+``` ts
+    showRetry (): boolean;
+```
+
+Params: none
+
+Returns: a boolean value determining if a the show retry button is required
+
+#### `retryConnection()`
+Function to retry connection to the selected implementation. It takes in the micLabel of the device in question and calls the `connect(label)` function of the corresponding implementation.
+    NOTE: This may be reworked
+
+Declaration:
+``` ts
+    retryConnection (micLabel: string): void;
+```
+
+Params:
+* `micLabel`: string - The label of the device in question to ensure proper functionality when bein passed into the `connect` function
+
+Returns: void
+
+#### `setRinging()`
```

Comment:
I can't remember if this is the case for all the vendors but I know some of them will do some logic to stop ringing while inside of the functions for answering or rejecting the calls.  Might need to revisit though to confirm missing the call rather than rejecting.  I believe I've seen the scenario where that still works though